GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -216,6 +216,32 @@ public virtual bool TryCreateToolsDirectory(out string upgraderToolPath, out str
             return false;
         }
 
+        public bool TryCleanup(out string error)
+        {
+            error = string.Empty;
+            foreach (Asset asset in this.newestRelease.Assets)
+            {
+                Exception exception;
+                if (!this.TryDeleteDownloadedAsset(asset, out exception))
+                {
+                    error += $"Could not delete {asset.LocalPath}. {exception.ToString()}.";
+                }
+            }
+
+            if (!string.IsNullOrEmpty(error))
+            {
+                return false;
+            }
+
+            error = null;
+            return true;
+        }
+
+        protected virtual bool TryDeleteDownloadedAsset(Asset asset, out Exception exception)

```

Comment:
Are you expecting this method to be more complicated in the future? Why not just call `TryDeleteFile` directly?

Reply:
(nevermind, I see you are overriding it in `MockProductUpgrader`)